src/cluster/index.js

```diff
@@ -389,19 +389,23 @@ module.exports = class Cluster {
     const topicConfigurations = {}
 
     const addDefaultOffset = topic => partition => {
-      const { fromBeginning } = topicConfigurations[topic]
-      return { ...partition, timestamp: this.defaultOffset({ fromBeginning }) }
+      const { timestamp } = topicConfigurations[topic]
+      return { ...partition, timestamp }
     }
 
     // Index all topics and partitions per leader (nodeId)
     for (const topicData of topics) {
-      const { topic, partitions, fromBeginning } = topicData
+      const { topic, partitions, fromBeginning, fromTimestamp } = topicData
       const partitionsPerLeader = this.findLeaderForPartitions(
         topic,
         partitions.map(p => p.partition)
       )
+      let timestamp = fromTimestamp
```

Comment:
I think this would be easier to read as a single expression, instead of mutating `timestamp`:

```js
const timestamp = fromTimestamp != null ? fromTimestamp : this.defaultOffset({ fromBeginning })
```